### ISSUE TITLE
Accessibility: `GridField` next/previous record links are problematic when disabled

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-framework#11818]
[ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/#namefromprohibited) Cannot use aria-label or aria-labelledby on elements and roles that prohibit naming.

In an edit form inside a gridfield, if there is no next or previous record to go to, "Go to previous/next record" link (bottom right of the screen) is represented by a `span`.

This has a few side effects:
1. It is not accessible with the keyboard (doesn't have `tab-index="0"`)
2. It may be skipped by screen readers in some cases
4. If it _is_ announced by screen readers it will be confusing, because you will hear "Go to previous record" but you can't interact with it, and there's no indication it's disabled.
3. It fails the above noted accessibility rule regarding `aria-labelledby`

## Acceptance criteria

- If this element _should_ be accessible with the keyboard:
    - [ ] use `<a>` instead of `<span>` 
    - [ ] add `pointer-events: none;` and `cursor: default;` CSS to disable its functionality
    - [ ] add a `aria-disabled="true"` attribute
- If this element should _not_ be accessible with the keyboard:
    - [ ] Either remove the `aria-label` attribute or give the span an appropriate `role`
    - [ ] change the `title` (and `aria-label` if keeping) text to indicate it's disabled (`aria-disabled` is being deprecated as a global state so we can't just chuck that on [unless it has a `role` that explicitly supports it](https://www.w3.org/TR/wai-aria-1.2/#aria-disabled))
- [ ] The changelog clearly calls out any changes to markup selectors so people can adjust their custom css/js as necessary

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/11892
- https://github.com/silverstripe/silverstripe-admin/pull/2045
- https://github.com/silverstripe/developer-docs/pull/834

**Comment by GuySartorelli** [comment]
PR merged
